### ISSUE TITLE
Windows Phone 8.1 (WinJS App) Cordova App

### ISSUE THREAD

**Issue by nirus** [pouchdb/pouchdb#5488]
### Environment (Node.js/browser/hybrid app/etc.)

Ionic Framework Hybrid Mobile App

### Browser/platform (Chrome/FF/Safari/Edge/iOS/Android/etc.)

Windows Phone 8.1 (WinJS App) Lumia 625

### Adapter (IndexedDB/WebSQL/LevelDB/etc.)

WebSQL

### Server (CouchDB/Cloudant/Couchbase/PouchDB Server/etc.)

(Write here.)

### Please describe your issue. Test cases (e.g. JSBin) are very helpful!

(Write here.)

Cordova Plugin from Microsoft for Web SQL: https://github.com/Microsoft/cordova-plugin-websql

I had developed Hybrid Mobile App using Pouch DB for Windows Phone.
Storage adapter used was 'websql' which failed without reason. Then i figured out and changed `Valid()` function logic which perfectly fixed the issue. Below is the code i modified in the library pouchdb.js.

```javascript
function valid() {
    var ret = false;
    if (typeof SQLitePlugin !== 'undefined') {
        ret = true;
    }else if (typeof openDatabase !== 'undefined') {
        ret = true;
    } else {
        ret = isValidWebSQL();
    }
    return ret;
}
```

or more cleaner code

```javascript
function valid() {

    if( (typeof SQLitePlugin !== 'undefined') || (typeof openDatabase !== 'undefined') || isValidWebSQL()) {
        return true;
    }
    return false;
}
```
This might help some one stuck. 

**Great plugin keep up the good work.** 

Sample App: [sampleApp.zip](https://github.com/pouchdb/pouchdb/files/376094/sampleApp.zip)

Request author to check the library on Windows Phone 8.1(Attached the project with issue). Please let me know your thoughts on this.

**Comment by nolanlawson** [comment]
I'm confused about your proposed fix because `isValidWebSQL()` should return true if `openDatabase` is defined, which is what the Microsoft WebSQL Plugin does: https://github.com/pouchdb/pouchdb/blob/ce73b62428d61225e23fa7c1515767b64bdd3eb9/packages/pouchdb-adapter-websql/src/validation.js#L16-L50 So I'm not sure why it's not working already.

Also it's unclear to me what your code actually changes, because I can't see the "before and after" of your code. Could you submit a pull request please?

**Comment by nolanlawson** [comment]
I have some Windows 8.1 Phones at work and can try to test this, but I'm really swamped for time these days so it probably won't be anytime soon. PRs would be much appreciated. Also keep in mind that our WebSQL validation logic has to check for odd iOS UIWebView edge cases, which is why it looks so convoluted.

**Comment by nirus** [comment]
I will test and let you know for v 5.4.5.

**Comment by nirus** [comment]
+1 - Fixed in 5.4.5

Good choice made for checking the `cordova` and returning `true`.
